src/read.c

```diff
@@ -3374,6 +3374,11 @@ static avifResult avifParseCondensedImageBox(avifMeta * meta, uint64_t rawOffset
     AVIF_CHECKERR(avifParseCodecConfiguration(&s, &colorCodecConfig, "coni", diag),
                   AVIF_RESULT_BMFF_PARSE_FAILED); // unsigned int(8) main_item_codec_config[];
 
+    // Make sure all extended_meta, metadata and coded chunks fit into the 'coni' box whose size is rawLen.
+    // There should be no missing nor unused byte.
+    AVIF_CHECKERR(avifROStreamOffset(&s) + extendedMetaSize + iccDataSize + alphaItemDataSize + colorItemDataSize + exifSize + xmpSize == rawLen,
```

Comment:
Nit: `rawLen` is the size of the stream `s`. So we don't need to use `rawLen` explicitly and can use a suitable avifROStream function instead, e.g., `avifROStreamRemainingBytes(&s) == extendedMetaSize + iccDataSize + alphaItemDataSize + colorItemDataSize + exifSize + xmpSize`.

Also, we should check for integer overflow in the sum `extendedMetaSize + iccDataSize + alphaItemDataSize + colorItemDataSize + exifSize + xmpSize`, unless some higher-level code guarantees that that sum cannot possibly overflow. This can be done in a separate pull request.